Trees/BreadthFirstTreeTraversal.js

```diff
@@ -17,7 +17,27 @@ class BinaryTree {
     this.traversal = []
   }
 
-  breadthFirst () {
+  breadthFirstIterative () {
+    this.traversal = []
+    if (this.root) {
+      this.traversal.push(this.root)
+    }
+    let currentSize = this.traversal.length
```

Comment:
Any reason why this needs to be a variable?

Reply:
You mean "currentSize"?

I wanted to manually update the variable myself so that the for-loop would 100% always check the value I expected. You wouldn't normally update the conditional dynamically in a for-loop (besides the index variable 'i'), so this was just to ensure I got the behavior I wanted without relying on some less obvious implementation detail regarding javascript.

Reply:
Hmm, I don't think list length is "a less obvious implementation detail"...